library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/mapper/MaskAllTextViewWireframeMapperTest.kt

```diff
@@ -40,27 +39,108 @@ internal class MaskAllTextViewWireframeMapperTest : BaseTextViewWireframeMapperT
     }
 
     override fun resolveTextValue(textView: TextView): String {
-        return String(CharArray(textView.text.length) { 'x' })
+        val text = textView.text
+        return String(
+            CharArray(text.length) {
+                val character = text[it]
+                if (character == ' ' || character == '\n') {
+                    character
+                } else {
+                    'x'
+                }
+            }
+        )
     }
 
     // endregion
 
     // region Unit tests
 
     @Test
-    fun `M resolve a TextWireframe with masked text W map() { TextView with text }`(forge: Forge) {
+    fun `M resolve a TextWireframe with masked text W map(){TextView, no newline or space}`(
+        forge: Forge
+    ) {
         // Given
-        val fakeText = forge.aString { 'x' }
-        val mockButton: TextView = forge.aMockView<Button>().apply {
-            whenever(this.text).thenReturn(forge.aString(fakeText.length))
+        val fakeExpectedText = forge.aString { 'x' }
+        val fakeText = forge.aString(fakeExpectedText.length) { forge.anAlphaNumericalChar() }
+        val mockTextView: TextView = forge.aMockView<TextView>().apply {
+            whenever(this.text).thenReturn(fakeText)
             whenever(this.typeface).thenReturn(mock())
         }
 
         // When
-        val textWireframe = testedTextWireframeMapper.map(mockButton, fakePixelDensity)
+        val textWireframe = testedTextWireframeMapper.map(mockTextView, fakePixelDensity)
 
         // Then
-        val expectedWireframe = mockButton.toTextWireframe().copy(text = fakeText)
+        val expectedWireframe = mockTextView.toTextWireframe().copy(text = fakeExpectedText)
```

Comment:
You should also have a test in the form 

```kotlin
val chunk1 = randomText
val chunk2 = randomText
val spacer = randomWhitespace // forge.forge.aWhitespaceString()
val input = chunk1 + spacer + chunk2
val expectedOutput = ("x" * chunk1.length) + spacer + ("x" * chunk2.length)
```

with this (I don't know if it exist in our repo

```kotlin
operator fun String.times(i: Int) {
    val builder = StringBuilder(i * length)
    repeat(i) {
        builder.append(this)
    }
}
```

Reply:
Good point